### ISSUE TITLE
Feature request: wait commands for rds cluster creation, rds cluster deletion, rds cluster snapshot creation

### ISSUE THREAD

**Issue by lancelakey** [aws/aws-cli#2394]
It'd be nice to have wait commands for rds cluster creation, rds cluster deletion, rds cluster snapshot creation

http://docs.aws.amazon.com/cli/latest/reference/rds/wait/index.html
From the documentation:
Available Commands
db-instance-available
db-instance-deleted
db-snapshot-completed

**Comment by kyleknap** [comment]
Seems reasonable. There are API's that will allow us to do that as well:
* http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBClusters.html
* http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DescribeDBClusterSnapshots.html

Marking as feature request

**Comment by xatnys** [comment]
This will be particularly useful for cross-region-replica Aurora clusters since cluster creation appears to take much longer, and [current guidance](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.Replication.CrossRegion.html) is to create the instances only after the cluster is in an `Available` state.

**Comment by ASayre** [comment]
Good Morning!

We're closing this issue here on GitHub, as part of our migration to [UserVoice](https://aws.uservoice.com/forums/598381-aws-command-line-interface) for feature requests involving the AWS CLI.

This will let us get the most important features to you, by making it easier to search for and show support for the features you care the most about, without diluting the conversation with bug reports.

As a quick UserVoice primer (if not already familiar): after an idea is posted, people can vote on the ideas, and the product team will be responding directly to the most popular suggestions.

We’ve imported existing feature requests from GitHub - Search for this issue there!

And don't worry, this issue will still exist on GitHub for posterity's sake.  As it’s a text-only import of the original post into UserVoice, we’ll still be keeping in mind the comments and discussion that already exist here on the GitHub issue.

GitHub will remain the channel for reporting bugs.  

Once again, this issue can now be found by searching for the title on: https://aws.uservoice.com/forums/598381-aws-command-line-interface 

-The AWS SDKs & Tools Team

**Comment by jamesls** [comment]
Based on community feedback, we have decided to return feature requests to GitHub issues.

**Comment by musch711** [comment]
Is there an update on this feature request?  I'm interested in this functionality as well.

**Comment by rahworkx** [comment]
I would like this feature to happen as well.

**Comment by shugigo** [comment]
+1

**Comment by dustin-rcg** [comment]
+1

**Comment by tim-finnigan** [comment]
The available rds wait commands has expanded since the original post:

- db-cluster-snapshot-available
- db-cluster-snapshot-deleted
- db-instance-available
- db-instance-deleted
- db-snapshot-available
- db-snapshot-completed
- db-snapshot-deleted

I reached out to the RDS team with the feature request to add these waiters:

- db-cluster-available
- db-cluster-deleted

Please note we are actively tracking this request here in our shared SDK repository: https://github.com/aws/aws-sdk/issues/137

Since this is a duplicate I will close this issue. Please refer to that issue for updates going forward.
